python_modules/dagster/dagster/core/asset_defs/assets_job.py

```diff
@@ -265,7 +265,9 @@ def build_deps(
             node_key = node_name
         deps[node_key] = {}
         assets_defs_by_node_handle[NodeHandle(alias, parent=None)] = assets_def
-        for input_name, asset_key in assets_def.asset_keys_by_input_name.items():
+        for input_name, asset_key in sorted(
+            assets_def.asset_keys_by_input_name.items(), key=lambda input: input[0]
```

Comment:
nit: you can unpack the tuple in the lambda args, so this can be written as

`key = lambda input_name, _: input_name`

which is a bit easier to read.

Reply:
Sadly unpacking a tuple within a lambda was removed between python 2 to 3 🥲